kmir/src/kmir/decoding.py

```diff
@@ -120,18 +121,30 @@ def _decode_memory_alloc_or_unable(
                 except KeyError:
                     return UnableToDecodeValue(f'Unknown pointee type: {pointee_ty}')
 
-                metadata = _metadata(pointee_type_info)
+                metadata_size = _metadata_size(pointee_type_info)
 
                 if len(data) == 8:
                     # single slim pointer (assumes usize == u64)
-                    return AllocRefValue(alloc_id=alloc_id, metadata=metadata)
+                    return AllocRefValue(
+                        alloc_id=alloc_id,
+                        metadata=Metadata(
+                            size=metadata_size,
+                            pointer_offset=0,
+                            origin_size=metadata_size,
+                        ),
```

Comment:
I made this assumption based on the previous version of `AllocRevValue.to_kast()`, I don't know if it makes sense though.

Reply:
I think this makes sense, when we decode we won't ever have an offset, and the `origin_size` should be the same as the `size` field IIUC.